File: module_login/src/main/java/cn/jack/module_login/mvvm/view/SplashActivity.java
package cn.jack.module_login.mvvm.view;

import android.os.Handler;

import cn.jack.library_arouter.manager.ArouterManager;
import cn.jack.module_login.R;
import cn.jack.module_login.databinding.ActivitysPlashBinding;
import jack.wrapper.base.mvvm.view.activity.BaseSimpleActiviy;

public class SplashActivity extends BaseSimpleActiviy<ActivitysPlashBinding> {

    private Handler mHandler = new Handler();

    @Override
    protected int setLayoutRes() {
        return R.layout.activitys_plash;
    }

    @Override
    public void prepareData() {
        super.prepareData();

        mBinding.ferrisWheelView.startAnimation();

        mHandler.postDelayed(new Runnable() {
            @Override
            public void run() {
                ArouterManager.getInstance().navigation2Login();
                finish();
            }
        },3000);

    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        mBinding.ferrisWheelView.stopAnimation();
    }

}

File: module_login/src/main/java/cn/jack/module_login/mvvm/view/SplashActivity.kt
package cn.jack.module_login.mvvm.view

import android.annotation.SuppressLint
import cn.jack.module_login.databinding.ModuleLoginActivitysPlashBinding
import com.jack.lib_base.base.view.BaseSimpleActivity
import com.jack.lib_base.ext.launchActivity

@SuppressLint("CustomSplashScreen")
class SplashActivity : BaseSimpleActivity<ModuleLoginActivitysPlashBinding>(ModuleLoginActivitysPlashBinding::inflate) {
    override fun prepareData() {
        super.prepareData()

        mBinding.ferrisWheelView.startAnimation()
        postDelayed({
            launchActivity(LoginActivity::class.java)
            finish()
        }, 3000)
    }

    override fun onDestroy() {
        super.onDestroy()
        mBinding.ferrisWheelView.stopAnimation()
    }
}